pkg/apis/core/helper/helpers.go

```diff
@@ -246,7 +251,7 @@ var standardResources = sets.NewString(
 
 // IsStandardResourceName returns true if the resource is known to the system
 func IsStandardResourceName(str string) bool {
-	return standardResources.Has(str) || IsQuotaHugePageResourceName(core.ResourceName(str))
+	return standardResources.Has(str) || IsQuotaHugePageResourceName(core.ResourceName(str)) || IsUlimitResourceName(str)
```

Comment:
are we adding more unqualified resource names? I thought the direction was pushing the other way (qualifying cpu, memory, etc, with `kubernetes.io/` prefixes)

Reply:
(maybe I'm misreading https://github.com/kubernetes/kubernetes/issues/50658#issuecomment-376983173)

Reply:
in any case, is this allowing setting unqualified resources in pod specs we didn't previously allow, as part of an alpha feature? if so, should it only be allowed when the feature is enabled? what are the implications if you set this and the backing runtime doesn't support it?

Reply:
@liggitt we are warning when the underlying runtime does not support it - https://github.com/kubernetes/kubernetes/pull/58904/commits/65e7d63e371e2d2aaedb2da32042d92509e077d2#diff-bf28da68f62a8df6e99e447c4351122dR913

we could elevate that to a fatal error?

Also, do you want me to add a check for `utilfeature.DefaultFeatureGate.Enabled(features.SupportUlimits)` here?

Reply:
> we are warning when the underlying runtime does not support it - 65e7d63#diff-bf28da68f62a8df6e99e447c4351122dR913

I don't have a good sense for the way unknown/unsupported resources have been handled in the past. Intuitively, if I declare my pod requires resource x and it is unsupported, I would expect my pod to not be run, and the status reflect that the resource I required wasn't supported.

> do you want me to add a check for utilfeature.DefaultFeatureGate.Enabled(features.SupportUlimits) here?

if we're calling this an alpha feature, and validation prevented pods from requesting this in the past, we should prevent the possibility of persisted pods with this resource until we're confident enough in the feature's stability/longevity to promote it

Reply:
ack, reworking so if the resource is needed and the feature is absent then we error out.